packages/components/src/flex/index.js

```diff
@@ -54,7 +61,9 @@ function FlexComponent(
 			ref={ ref }
 			gap={ gap }
 			justify={ justify }
-			isReversed={ isReversed }
+			isReversed={ reversed }
+			// @ts-ignore
+			direction={ simplifiedDirection }
```

Comment:
TS was angry about this potentially being `undefined` when we have noted it as a string.  But if we look at the code above, it would always be a string.  So `@ts-ignore` for now unless there are better suggestions.

Reply:
`simplifiedDirection` is a string beucase of `direction.slice( 0, -8)`. That's why TS is raging. I'm not sure if it is possible to force a type on a specific variable with JSDoc. `simplifiedDirection as FlexDirection` could fix the problem in TS.

I think you are better off with an explicit function types-wise:
```js
/** @typedef {import('react').CSSProperties['flexDirection'] | 'row' | 'column'} FlexDirection */
^^^ put this into `/* eslint-disable jsdoc/valid-types */` above OwnProps

 * @property {FlexDirection} direction Sets the flex-direction property.
^^^
and replace the type in OwnProps 


/**
 *
 * @param {FlexDirection} direction
 * @return {FlexDirection} direction
 */
function simplifyDirection( direction ) {
	if ( direction === 'column-reverse' ) {
		return 'column';
	}

	if ( direction === 'row-reverse' ) {
		return 'row';
	}

	return direction;
}
```

then you can use it like 
```js
const simplifiedDirection = simplifyDirection( direction );
```

and you can remove the @ts-ignore